agent/consul/state/catalog.go

```diff
@@ -1086,6 +1086,43 @@ func (s *Store) EnsureCheck(idx uint64, hc *structs.HealthCheck) error {
 	return nil
 }
 
+// This ensure all connected services to a check are updated properly
+func (s *Store) alterServicesFromCheck(tx *memdb.Txn, idx uint64, hc *structs.HealthCheck, checkServiceExists bool) error {
```

Comment:
Go style is to start the comment with the name of the method and write full sentence:

```go
// alterServicesFromCheck ensures ... are updated properly.
func (s *Store) alterServicesFromCheck(...)
```

Also the name confuses me: it's names "alter services" yet it actually updates the service _index_ right not the actual service record?

Reply:
DONE in next patch